src/backend/distributed/deparser/qualify_statistics_stmt.c

```diff
@@ -48,3 +48,29 @@ QualifyCreateStatisticsStmt(Node *node)
 		stmt->defnames = MakeNameListFromRangeVar(stat);
 	}
 }
+
+
+/*
+ * QualifyDropStatisticsStmt qualifies DropStmt's with schema name for
+ * DROP STATISTICS statements.
+ */
+void
+QualifyDropStatisticsStmt(Node *node)
+{
+	DropStmt *dropStatisticsStmt = castNode(DropStmt, node);
+	Assert(dropStatisticsStmt->removeType == OBJECT_STATISTIC_EXT);
+
+	List *objectNameList = NULL;
+	foreach_ptr(objectNameList, dropStatisticsStmt->objects)
+	{
+		RangeVar *stat = makeRangeVarFromNameList(objectNameList);
+
+		if (stat->schemaname == NULL)
+		{
+			Oid schemaOid = RangeVarGetCreationNamespace(stat);
+			stat->schemaname = get_namespace_name(schemaOid);
+
+			*objectNameList = *MakeNameListFromRangeVar(stat);
```

Comment:
similar pattern, would be nicer to construct a new list of lists (keeping entries that have a schema name the same)